spec/unit/actions/synchronize_user_filters_spec.rb

```diff
@@ -18,54 +18,57 @@
     allow(project).to receive(:pid).and_return('123456789')
     allow(user).to receive(:login).and_return('my_login')
     allow(GoodData::Helpers::DataSource).to receive(:new).and_return(data_source)
+    allow(logger).to receive(:warn)
   end
 
   context 'when multiple_projects_column not specified' do
     before do
       allow(project).to receive(:metadata).and_return({})
       allow(project).to receive(:uri)
-      allow(domain).to receive(:clients).and_return(organization)
       allow(data_source).to receive(:realize)
-      allow(organization).to receive(:find)
+      allow(organization).to receive(:id).and_return('client123')
+      allow(organization).to receive(:project).and_return(project)
     end
     context 'when mode requires client_id' do
-      let(:mode) { 'sync_one_project_based_on_pid' }
+      before do
+        allow(domain).to receive(:clients).and_return(organization)
+      end
+      let(:mode) { 'sync_multiple_projects_based_on_custom_id' }
       let(:params) do
         params = {
           GDC_GD_CLIENT: client,
           input_source: 'foo',
           domain: 'bar',
           filters_config: { labels: [] },
-          sync_mode: mode
+          sync_mode: mode,
+          gdc_logger: logger
         }
         GoodData::LCM2.convert_to_smart_hash(params)
       end
 
       before do
-        allow(File).to receive(:open).and_return("project_id\n123456789")
+        allow(File).to receive(:open).and_return("client_id\n123456789")
       end
 
-      it 'uses project_id column' do
-        expect(GoodData::UserFilterBuilder).to receive(:get_filters) do |filters, _|
-          CSV.foreach(filters, headers: true, return_headers: false) do |row|
-            expect(row['project_id']).to eq '123456789'
-          end
-          []
-        end
+      it 'uses client_id column' do
+        expect(domain).to receive(:clients).with('123456789', nil)
         subject.class.call(params)
       end
 
       context 'when mode is sync_domain_client_workspaces' do
         let(:mode) { 'sync_domain_client_workspaces' }
         let(:clients) { [] }
         before do
-          allow(domain).to receive(:clients).with(no_args).and_return(clients)
+          allow(domain).to receive(:clients).with(:all, nil).and_return([organization])
+          allow(domain).to receive(:clients).with(123456789).and_return(organization)
```

Comment:
Style/NumericLiterals: Separate every 3 digits in the integer portion of a number with underscores(_).